src/write_gcm.c

```diff
@@ -4158,6 +4158,27 @@ static int wg_write(const data_set_t *ds, const value_list_t *vl,
     }
     --queue->size;
     to_remove->next = NULL;
+    char metadata[8192];
+    metadata[0] = '{';
+    char *meta_ptr = metadata + 1;
+    int meta_size = sizeof(metadata) - 1;
```

Comment:
It's precisely this kind of "should 1 be 2" FUD that caused me to write the method `bufprintf` in the first place.

Also note that you got the sense of your ternary operator backwards: the first time through the loop is when you want the empty string. I usually write that differently anyway (see below).

Hope you're all well. Lunch sometime?

If it were up to me, I'd change the above to
```
char metadata[8192];
char *meta_ptr = metadata;
size_t meta_size = sizeof(metadata);
bufprintf(&meta_ptr, &meta_size, "{");
const char *comma = "";
for (int i = 0; i < to_remove->key.num_metadata_entries; ++i) {
  bufprintf(&meta_ptr, &meta_size, "%s'%s' = '%s'",
          comma,
          to_remove->key.metadata_entries[i].key,
          to_remove->key.metadata_entries[i].value.value_text);
  comma = ", ";
}
bufprintf(&meta_ptr, &meta_size, "}");
```